src/Workspaces/Remote/ServiceHub/Services/BrokeredServiceBase.cs

```diff
@@ -86,7 +87,7 @@ protected async ValueTask<T> RunServiceAsync<T>(Func<CancellationToken, ValueTas
         protected async ValueTask RunServiceAsync(Func<CancellationToken, ValueTask> implementation, CancellationToken cancellationToken)
         {
             WorkspaceManager.SolutionAssetCache.UpdateLastActivityTime();
-            using var _ = LinkToken(ref cancellationToken);
```

Comment:
@tmat This was changing the cancellation token which was subsequently passed to `ReportWithoutCrashUnlessCanceledAndPropagate`.

Reply:
That was definitely intentional. The combined one is not used above.

Reply:
We want to cancel the processing of the service if either the client cancellation token is signaled or the client disconnects.